lib/versions.ps1

```diff
@@ -1,53 +1,258 @@
+    $splitDifferenceVersion = @(SplitVersion -Version $DifferenceVersion -Delimiter $Delimiter)
+
+    # Nightly versions are always equal
+    if ($splitReferenceVersion[0] -eq 'nightly' -and $splitDifferenceVersion[0] -eq 'nightly') {
+        return 0
+    }
+
+    for ($i = 0; $i -lt [Math]::Max($splitReferenceVersion.Length, $splitDifferenceVersion.Length); $i++) {
+        # '1.1-alpha' is less then '1.1'
+        if ($i -ge $splitReferenceVersion.Length) {
+            if ($splitDifferenceVersion[$i] -match 'alpha|beta|rc|pre') {
+                return -1
+            } else {
+                return 1
+            }
+        }
+        # '1.1' is greater then '1.1-beta'
+        if ($i -ge $splitDifferenceVersion.Length) {
+            if ($splitReferenceVersion[$i] -match 'alpha|beta|rc|pre') {
+                return 1
+            } else {
+                return -1
+            }
         }
 
-        if($ver_a[$i] -gt $ver_b[$i]) { return 1; }
-        if($ver_a[$i] -lt $ver_b[$i]) { return -1; }
+        # If some parts of versions have '.', compare them with delimiter '.'
+        if (($splitReferenceVersion[$i] -match '\.') -or ($splitDifferenceVersion[$i] -match '\.')) {
+            $Result = Compare-Version -ReferenceVersion $splitReferenceVersion[$i] -DifferenceVersion $splitDifferenceVersion[$i] -Delimiter '.'
+            # If the parts are equal, continue to next part, otherwise return
+            if ($Result -ne 0) {
+                return $Result
+            } else {
+                continue
+            }
+        }
+
+        # Don't try to compare [Long] to [String]
+        if ($null -ne $splitReferenceVersion[$i] -and $null -ne $splitDifferenceVersion[$i]) {
+            if ($splitReferenceVersion[$i] -is [String] -and $splitDifferenceVersion[$i] -isnot [String]) {
+                $splitDifferenceVersion[$i] = "$($splitDifferenceVersion[$i])"
+            }
+            if ($splitDifferenceVersion[$i] -is [String] -and $splitReferenceVersion[$i] -isnot [String]) {
+                $splitReferenceVersion[$i] = "$($splitReferenceVersion[$i])"
+            }
+        }
+
+        # Compare [String] or [Long]
+        if ($splitDifferenceVersion[$i] -gt $splitReferenceVersion[$i]) {
+            return 1
+        }
+        if ($splitDifferenceVersion[$i] -lt $splitReferenceVersion[$i]) {
+            return -1
+        }
     }
-    if($ver_b.length -gt $ver_a.length) { return -1 }
-    return 0
 }
 
+# Helper function
+function SplitVersion {
+    <#
+    .SYNOPSIS
+        Split version by Delimiter, convert number string to number, and separate letters from numbers
+    .PARAMETER Version
+        Specifies a version
+    .PARAMETER Delimiter
+        Specifies the delimiter of version (Literal)
+    #>
+    param (
+        [String]
+        $Version,
+        [String]
+        $Delimiter = '-'
+    )
+    $Version = $Version -replace '[a-zA-Z]+', "$Delimiter$&$Delimiter"
```

Comment:
What is this for?

Reply:
If `Delimiter` is `-` (or `.`), `1.3a1` will be replaced by `1.3-a-1` (`1.3.a.1`).

Thus letters will be seperated from numbers.